### PR TITLE
Add ability to recover config from conn.

### PR DESCRIPTION
**Pull request by aaspring**

### CODE REVIEW

**Comment by zakird** [comment]
lgtm

File: ztools/ztls/common.go [hunk 1]
```diff
@@ -194,13 +195,31 @@ type ConnectionState struct {
...
+
+	return "unknown"
+}
+
+func (authType *ClientAuthType) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + authType.String() + `"`), nil
+}
+
+func (authType *ClientAuthType) UnmarshalJSON(b []byte) error {
+	panic("unimplemented")
```
Comment (dadrian) [inline ztools/ztls/common.go]: :(

File: ztools/ztls/ztls_names.go [hunk 2]
```diff
@@ -380,6 +381,14 @@ func init() {
...
 	cipherSuiteNames[0xFF03] = "SSL_EN_RC2_128_CBC_WITH_MD5"
 	cipherSuiteNames[0xFF85] = "OP_PCL_TLS10_AES_128_CBC_SHA512"
+
+	// Name-value paires *are* not standardized, only dereferenced for JSON output
+	clientAuthTypeNames = make(map[int]string)
+	clientAuthTypeNames[0] = "NoClientCert"
+	clientAuthTypeNames[1] = "RequestClientCert"
+	clientAuthTypeNames[2] = "RequireAnyClientCert"
+	clientAuthTypeNames[3] = "VerifyClientCertIfGiven"
+	clientAuthTypeNames[4] = "RequireAndVerifyClientCert"
```
Comment (dadrian) [inline ztools/ztls/ztls_names.go]: What are these names coming from? Are these just a Golang abstraction for how to handle client certs?

Comment (aaspring) [inline ztools/ztls/ztls_names.go]: Yes, these are Golang abstractions from ztls/common.go. The actual values were previously an iota and have not meaning other than to be part of an enum-style class.

image

**Comment by zakird** [comment]
@aaspring this conflicts. Once that's resolved, this can be merged.

**Comment by aaspring** [comment]
Working on it now.

**Pull request merged**